src/control/Control.cpp

```diff
@@ -1523,50 +1529,43 @@ void Control::changePageBackgroundColor()
 	}
 }
 
-void Control::calcZoomFitSize()
+void Control::setViewPairedPages(bool enabled)
 {
 	XOJ_CHECK_TYPE(Control);
 
-	if (this->doc && this->win)
-	{
-		PageRef p = getCurrentPage();
-		if (!p.isValid())
-		{
-			return;
-		}
-		double page_width = p->getWidth() + 20;
-
-		Rectangle widget_rect = zoom->getVisibleRect();
-		double factor = widget_rect.width / page_width;
-		zoom->setZoomFitValue(factor);
-	}
-}
-
-void Control::setZoomFitButton(bool zoomFit)
-{
-	XOJ_CHECK_TYPE(Control);
-
-	fireActionSelected(GROUP_ZOOM_FIT, zoomFit ? ACTION_ZOOM_FIT : ACTION_NOT_SELECTED);
-}
-
-void Control::setViewPairedPages(bool pairedPages)
-{
-	XOJ_CHECK_TYPE(Control);
-
-	settings->setShowPairedPages(pairedPages);
-	fireActionSelected(GROUP_PAIRED_PAGES, pairedPages ? ACTION_VIEW_PAIRED_PAGES : ACTION_NOT_SELECTED);
+	settings->setShowPairedPages(enabled);
+	fireActionSelected(GROUP_PAIRED_PAGES, enabled ? ACTION_VIEW_PAIRED_PAGES : ACTION_NOT_SELECTED);
 
 	int currentPage = getCurrentPageNo();
 	win->getXournal()->layoutPages();
 	scrollHandler->scrollToPage(currentPage);
 }
 
-void Control::setViewPresentationMode(bool presentationMode)
+void Control::setViewPresentationMode(bool enabled)
 {
 	XOJ_CHECK_TYPE(Control);
 
-	settings->setPresentationMode(presentationMode);
-	fireActionSelected(GROUP_PRESENTATION_MODE, presentationMode ? ACTION_VIEW_PRESENTATION_MODE : ACTION_NOT_SELECTED);
+	if(enabled)
+	{
+		// Disable PresentationMode if ZoomValue could not calculated correctly
+		zoom->updateZoomPresentationValue();
+		//TODO: write out to settings previous layout
+		setViewPairedPages(false);
+		setViewRows(0);
+		setViewColumns(0);
```

Comment:
Shouldn't that be `setViewColumns(1)` so we have one column for pages?

Reply:
LittleHuba is correct but now that you've merged in #1037 you should not change the layout settings here.
You should remove the calls to  setViewRows, setViewColumns and setViewPairedPages.
PR 1037 did not configure the Paired Pages so you'll have to remedy that by changing   LayoutMapper::configureForPresentation() so that line
`	bool isPairedPages = settings->isShowPairedPages();`
instead reads:
`	bool isPairedPages = false;`

Reply:
When the user leaves presentation mode the layout will revert back to using the settings.
Because Presentation mode ignores the users layout settings we should also grey out the layout settings ( View->Layout and view->Cols/Rows ) in the menu when presentation mode is active. 
_Can do in another PR._

Reply:
@JJones780 we should do this in another PR as this one is already heavy enough. Additionally it is coming up with Merge Conflicts where we should not introduce another feature in fear of even more conflicts.

Reply:
I disabled the menuitems now, this should be enough.